### DISCUSSION TITLE
Configuration file?

### DISCUSSION THREAD

**Discussion by hatamiarash7** [projectdiscovery/httpx#1737] [Q&A]
Hey there 

How can I configure the httpx using a configuration file? I can see that the default path is `$HOME/.config/httpx/config.yaml` but I can't find any template or instruction to create this file. Is there any documentation for that?

**Comment by alpineyahoo** [comment]
I have the same question. I tried to set `silent: true` in `~/.config/httpx/config.yaml`, but it failed to load.

**Comment by GeorginaReeder** [comment]
Thanks for this - tagging @dwisiswant0 to take a look!

**Comment by l0pens** [comment]
hi, 👋 

can we convert this discussion as an issue:  
- goal/task: 💡 add to readme file a simple example on how to configure httpx yaml file `~/.config/httpx/config.yaml`

---
My 2 cents: 

- Without config, a default one is generated. That's fine.

But for example a concrete `auth-config` definition is unclear (at least for me) .. even if the httpx issue message is very helpful (`[asnmap-api] missing or invalid api key (get free api key & configure it from https://cloud.projectdiscovery.io/?ref=api_key)`) :)

NB: my need: use api key but won't interactive mode (and using PDCP_API_KEY env variable dont work as expected : stop after login?)
````bash
$ #~/.config/http/config.yaml with "auth: true" // v1.7.0
$ export PDCP_API_KEY="<get yours for free at https://cloud.projectdiscovery.io/?ref=api_key>"
$ cat domains.txt | httpx -asn

    __    __  __       _  __
   / /_  / /_/ /_____ | |/ /
  / __ \/ __/ __/ __ \|   /
 / / / / /_/ /_/ /_/ /   |
/_/ /_/\__/\__/ .___/_/|_|
             /_/

                projectdiscovery.io

[INF] You are logged in as (@REDACTED)
$ 
````

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
